selftests/functional/test_nrunner_interface.py

```diff
@@ -73,3 +73,15 @@ def test_runnable_run_recipe_specific_kind(self):
         cmd = f"{runner} runnable-run-recipe {recipe}"
         result = process.run(cmd, ignore_status=True)
         self.assertEqual(result.exit_status, 0)
+
+    def test_task_run_no_args(self):
+        cmd = f"{self.get_runner()} task-run"
+        result = process.run(cmd, ignore_status=True)
+        self.assertEqual(result.exit_status, 2)
+
+    def test_task_run_identifier_only(self):
```

Comment:
Because of this test I started thinking, If this is the right behavior. IIUIC avocado runner can't  run task just based on identifier so IMO the command `avocado-runner-avocado-instrumented task-run -i some_identifier` should exit with code 2. My concern is not related to these tests, but to the `task-run` behavior itself.

Reply:
This is consistent with the `runnable-run` approach, although in its case.  Basically, when a runner follows the `avocado-runner-$type` name, the `-k|--kind` parameter is deduced from `$type`.

So, you'll "bare" `avocado-runner task-run -i some_identifier` returning 2 (because it's missing a kind), and `avocado-runner-$type task-run -i identifier` passing, because all mandatory info is given.

Can you better explain the part of the behavior you'd suggest a change on?

Reply:
Yes, I understand the difference between `avocado-runner` and `avocado-runner-$type` and that is working fine, I don't have a problem with it. What I meant is that IMO the task can't be run just based on the `kind` and `identifier` without `uri`.  So I think the `avocado-runner task-run -i some_identifier` should return 2 because it's missing a kind and an uri and `avocado-runner-$type task-run -i identifier` should return 2 because it's missing an uri.

Reply:
OK, now I understand your concern, but I don't quite agree with it.  For instance:

```
$ avocado-runner-noop runnable-run
{'status': 'started', 'time': 1828344.398746594}
{'result': 'pass', 'status': 'finished', 'time': 1828344.39876635}
$ echo $?
0
```

Adding a task to this runnable execution:

```
$ avocado-runner-noop task-run -i 1-noop
{'status': 'started', 'time': 1828408.281513233, 'output_dir': '/tmp/.avocado-task-t83r7eo8', 'id': '1-noop'}
{'result': 'pass', 'status': 'finished', 'time': 1828408.281535025, 'id': '1-noop'}
$ echo $?
0
```

Making the URI mandatory would have to be done at the runnable level to be consistent.  There are kinds of tests (`noop`, `dry-run`) that don't require an URI, so why would we make it mandatory?

Reply:
Yes, you are right that this can't be done in general for all kinds. I forgot about those as `noop` and `dry-run`. But I still think we should fix this, because for some runners the `uri` is mandatory, and it seems strange to me to have for example `avocado-runner-avocado-instrumented task-run -i some_identifier` with exit 0.